### DISCUSSION TITLE
Numerics automatically returned in scientific notation. How to make that optional?

### DISCUSSION THREAD

**Discussion by seperman** [psycopg/psycopg2#1480] [Q&A]
Hello,
The numeric types are automatically returned by psycopg2 in scientific notation. This is not desired in all cases. How can we make the scientific notation optional?
Thanks

**Comment by dvarrazzo** [comment]
"scientific notation" is a representation. Psycopg returns just ordinary Python float.

I assume you are seeing the way PostgreSQL returns those numbers. Can you make an example please?

**Comment by seperman** [comment]
It is actually returning strings for me. I do a fetchall and I get a list of tuples of strings that have my decimals in scientific notation. I did check the type too. It is str.

Sep Dehpour

**Reply by dvarrazzo** [reply]
If your data is returned as strings, then probably it's not a standard Postgres numeric type, or you have manipulated the standard typecasters.

What oid is the type? There are [examples in the docs](https://www.psycopg.org/docs/advanced.html#type-casting-of-sql-types-into-python-objects) about finding the kind of a postgres data type.

**Comment by seperman** [comment]
I will add an example once I get back in front of my computer.

Sep Dehpour

**Comment by seperman** [comment]
The weird part is that it is not consistent. Now it returns them as floats as expected. I will update it here once it returns them in the scientific notation again.
